modules/collect/src/main/java/com/opengamma/strata/collect/MapStream.java

```diff
@@ -0,0 +1,391 @@
+   * @return a stream including the entries for which the predicate function returned true
+   */
+  public MapStream<K, V> filterValues(Predicate<? super V> predicate) {
+    return wrap(underlying.filter(e -> predicate.test(e.getValue())));
+  }
+
+  /**
+   * Transforms the keys in the stream by applying a mapper function to each key.
+   * <p>
+   * The values are unchanged.
+   *
+   * @param mapper  a mapper function whose return value is used as the new key
+   * @param <R>  the type of the new keys
+   * @return a stream of entries with the keys transformed and the values unchanged
+   */
+  public <R> MapStream<R, V> mapKeys(Function<? super K, ? extends R> mapper) {
+    return wrap(underlying.map(e -> entry(mapper.apply(e.getKey()), e.getValue())));
+  }
+
+  /**
+   * Transforms the keys in the stream by applying a mapper function to each key and value.
+   * <p>
+   * The values are unchanged.
+   *
+   * @param mapper  a mapper function whose return value is used as the new key
+   * @param <R>  the type of the new keys
+   * @return a stream of entries with the keys transformed and the values unchanged
+   */
+  public <R> MapStream<R, V> mapKeys(BiFunction<? super K, ? super V, ? extends R> mapper) {
+    return wrap(underlying.map(e -> entry(mapper.apply(e.getKey(), e.getValue()), e.getValue())));
+  }
+
+  /**
+   * Transforms the values in the stream by applying a mapper function to each value.
+   * <p>
+   * The keys are unchanged.
+   *
+   * @param mapper  a mapper function whose return value is used as the new value
+   * @param <R>  the type of the new values
+   * @return a stream of entries with the values transformed and the keys unchanged
+   */
+  public <R> MapStream<K, R> mapValues(Function<? super V, ? extends R> mapper) {
+    return wrap(underlying.map(e -> entry(e.getKey(), mapper.apply(e.getValue()))));
+  }
+
+  /**
+   * Transforms the values in the stream by applying a mapper function to each key and value.
+   * <p>
+   * The keys are unchanged.
+   *
+   * @param mapper  a mapper function whose return value is used as the new value.
+   * @param <R>  the type of the new values
+   * @return a stream of entries with the values transformed and the keys unchanged
+   */
+  public <R> MapStream<K, R> mapValues(BiFunction<? super K, ? super V, ? extends R> mapper) {
+    return wrap(underlying.map(e -> entry(e.getKey(), mapper.apply(e.getKey(), e.getValue()))));
+  }
+
+  /**
+   * Transforms the entries in the stream by applying a mapper function to each key and value.
+   *
+   * @param mapper  a mapper function whose return values are included in the new stream
+   * @param <R>  the type of elements in the new stream
+   * @return a stream containing the values returned from the mapper function
+   */
+  public <R> Stream<R> map(BiFunction<? super K, ? super V, ? extends R> mapper) {
+    return underlying.map(e -> mapper.apply(e.getKey(), e.getValue()));
+  }
```

Comment:
Add `mapToValues()` and `mapToKeys()`. These would be useful if you need to filter by key but then only need the values.